### PR TITLE
feat: set feature flag system in place

### LINKED ISSUE #329: Set feature flag system in place for working on the UI
- Set feature flag system in place for working on the UI

### PR DESCRIPTION
**Pull request by cka-y**
**Summary:**  
Closes #329 
This PR integrates a feature flag system by leveraging Firebase Remote Config capabilities. 

To add a new feature flag, simply append its name and default value to `interface/RemoteConfig.ts`. The Firebase console allows for the remote modification of these flags, overriding their default values as soon as the cache expires.

**Expected Behavior:**  
Feature flags serve as switches for enabling or disabling application features. The implementation facilitates server-side updates and control over feature availability directly from the Firebase console.

**Note about update rate of feature flags:**  
In the production environment, the feature flag update rate is configured to 12 hours, adhering to Firebase's best practice recommendations to prevent throttling. For development and QA environments, this rate is reduced to 5 minutes, allowing for rapid testing and iteration. More details on throttling can be found in the [Firebase documentation](https://firebase.google.com/docs/remote-config/get-started?hl=en&authuser=0&_gl=1*jbxp92*_ga*NjEwMjk3NjY5LjE3MTAzNjIzMTg.*_ga_CW55HF8NVT*MTcxMDM2MjMxOC4xLjEuMTcxMDM2MjMyNC41NS4wLjA.&platform=web#throttling).

**Testing Instructions:**  
To test the functionality of a feature flag:
1. Navigate to the Firebase Console.
2. Access the `mobility-feeds-dev` project.
3. Select Build > Remote Config from the menu.
4. Locate the `enableGithubSSO` configuration and set its value to `false`.
5. Allow up to 5 minutes for the cache to update.
6. Upon visiting the login or sign-up page, the option to use GitHub for authentication should be disabled.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: web-app/src/app/screens/SignUp.tsx [hunk 1]
```diff
@@ -372,30 +377,34 @@ export default function SignUp(): React.ReactElement {
...
-          sx={{ mb: 2 }}
-          startIcon={<GitHubIcon />}
-          className='sso-button'
-          onClick={() => {
-            signInWithProvider(OauthProvider.Github);
-          }}
-        >
-          Sign Up With GitHub
-        </Button>
+        {enableGoogleSSO && (
```
Comment (davidgamez) [inline web-app/src/app/screens/SignUp.tsx:380]: Let's revert this change before merging.

File: web-app/src/app/screens/SignIn.tsx [hunk 2]
```diff
@@ -260,30 +265,34 @@ export default function SignIn(): React.ReactElement {
...
-          className='sso-button'
-          sx={{ mb: 2 }}
-          startIcon={<GitHubIcon />}
-          onClick={() => {
-            signInWithProvider(OauthProvider.Github);
-          }}
-        >
-          Sign In With GitHub
-        </Button>
+        {enableGoogleSSO && (
```
Comment (davidgamez) [inline web-app/src/app/screens/SignIn.tsx:268]: Let's revert this change before merging

File: web-app/src/app/router/Router.tsx [hunk 3]
```diff
@@ -15,36 +15,39 @@ import Contribute from '../screens/Contribute';
...
-        <Route path='verify-email' element={<PostRegistration />} />
-      </Route>
-      <Route path='forgot-password' element={<ForgotPassword />} />
-      <Route path='faq' element={<FAQ />} />
-      <Route path='about' element={<About />} />
-      <Route path='contribute' element={<Contribute />} />
-      <Route path='privacy-policy' element={<PrivacyPolicy />} />
-      <Route path='terms-and-conditions' element={<TermsAndConditions />} />
-    </Routes>
+    <RemoteConfigProvider>
```
Comment (davidgamez) [inline web-app/src/app/router/Router.tsx:22]: [question]: Can we move this provider to _App.tsx_ instead of the router? It will be a more natural place to look for it.

File: web-app/src/app/interface/RemoteConfig.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { remoteConfig } from '../../firebase';
+export type RemoteConfigValues = Record<string, string | number | boolean>;
+
+// Add default values for remote config here
+export const defaultRemoteConfigValues: RemoteConfigValues = {
+  enableGoogleSSO: true,
+  enableGithubSSO: true,
```
Comment (davidgamez) [inline web-app/src/app/interface/RemoteConfig.ts:7]: We can remove these flags used for testing and add _enableAppleSSO_ as it will be used right after this PR is merged

File: web-app/src/app/screens/SignUp.tsx [hunk 5]
```diff
@@ -56,6 +57,10 @@ export default function SignUp(): React.ReactElement {
   const signUpError = useSelector(selectSignUpError);
   const userProfileStatus = useSelector(selectUserProfileStatus);
   const [isSubmitted, setIsSubmitted] = React.useState(false);
+  const { config } = useRemoteConfig();
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const enableAppleSSO = config.enableAppleSSO as boolean;
```
Comment (cka-y) [inline web-app/src/app/screens/SignUp.tsx:63]: leaving this here for usage after PR is merged

File: web-app/src/app/screens/SignUp.tsx (hunk 1 above)
Comment (cka-y) [reply web-app/src/app/screens/SignUp.tsx]: ✅

File: web-app/src/app/screens/SignIn.tsx (hunk 2 above)
Comment (cka-y) [reply web-app/src/app/screens/SignIn.tsx]: ✅

File: web-app/src/app/router/Router.tsx (hunk 3 above)
Comment (cka-y) [reply web-app/src/app/router/Router.tsx]: moved ✅

File: web-app/src/app/interface/RemoteConfig.ts (hunk 4 above)
Comment (cka-y) [reply web-app/src/app/interface/RemoteConfig.ts]: ✅

**Review by davidgamez — APPROVED** [review]
LGTM! 💪
